archey/test/test_archey_wan_ip.py

```diff
@@ -17,16 +17,16 @@ class TestWanIpEntry(unittest.TestCase):
     @patch(
```

Comment:
Sorry again for those noises (IPv4 and IPv6 logic order has been inverted).

Reply:
Within this noise (😆) maybe we should look at changing our configuration patches to something like:
```python
@patch.dict(
    LanIp()._configuration._config,
    values={
        'some_config_section': {
            'needed_key': None,
            'testing_value': True
        }
    }
)
```
...since we now have the single instance, and this makes our tests less fragile (for example you had to swap around the patch `side_effect`s here for changed code but no changed behaviour)

Reply:
Sorry, just thought about that again and what would probably be more correct is to just mock out the entire configuration with a dict for a test like so:
```python
@patch(
    archey.entry.Configuration,
    return_value={
        'some_config_section': {
            'needed_key': None,
            'testing_value': True
        }
    }
)
```

...or, maybe mock `_config` only, by passing in the mock after patching to each test method and setting `configuration_mock.return_value._config` (so tests *fail* if `__get__` is used on `self._configuration`)... Either way, this has the advantage of not relying on the singleton during the tests - if a test somehow changes its contents all other tests would be unaffected (which I had fun with in the last PR I tried.... 🤦)

Reply:
Agreed ! Maybe we could look forward to enhance configuration patching behavior in another WIP or so ? :slightly_smiling_face:

Reply:
Yeah, I think it's a good candidate for WIP branch/PR once this is merged 👍

Reply:
So let's go then 👌